rust/src/ldap/ldap.rs

```diff
@@ -0,0 +1,426 @@
+                Ok((rem, msg)) => {
+                    start = rem;
+                    let mut tx = self.new_tx();
+                    let request = if let Ok(req) = LdapMessage::try_from(msg) {
+                        req
+                    } else {
+                        return AppLayerResult::err();
+                    };
+                    tx.request = Some(request);
+                    self.transactions.push_back(tx);
+                }
+                Err(nom::Err::Incomplete(_)) => {
+                    // Not enough data. This parser doesn't give us a good indication
+                    // of how much data is missing so just ask for one more byte so the
+                    // parse is called as soon as more data is received.
+                    let consumed = input.len() - start.len();
+                    let needed = start.len() + 1;
+                    return AppLayerResult::incomplete(consumed as u32, needed as u32);
+                }
+                Err(_) => {
+                    return AppLayerResult::err();
+                }
+            }
+        }
+
+        // Input was fully consumed.
+        return AppLayerResult::ok();
+    }
+
+    fn parse_response(&mut self, input: &[u8]) -> AppLayerResult {
+        // We're not interested in empty responses.
+        if input.is_empty() {
+            return AppLayerResult::ok();
+        }
+
+        if self.response_gap {
+            if !probe(input) {
+                // The parser now needs to decide what to do as we are not in sync.
+                // For this ldap, we'll just try again next time.
+                return AppLayerResult::ok();
+            }
+
+            // It looks like we're in sync with a message header, clear gap
+            // state and keep parsing.
+            self.response_gap = false;
+        }
+        let mut start = input;
+        while !start.is_empty() {
+            match LdapParserMessage::from_ber(&start) {
+                Ok((rem, msg)) => {
+                    start = rem;
+
+                    let response = if let Ok(resp) = LdapMessage::try_from(msg) {
+                        resp
+                    } else {
+                        return AppLayerResult::err();
+                    };
+                    if let Some(tx) = self.find_request(response.message_id) {
+                        tx.responses.push_back(response.clone());
+                    }
+                }
+                Err(nom::Err::Incomplete(_)) => {
+                    let consumed = input.len() - start.len();
+                    let needed = start.len() + 1;
+                    return AppLayerResult::incomplete(consumed as u32, needed as u32);
+                }
+                Err(_) => {
+                    return AppLayerResult::err();
+                }
+            }
+        }
+
+        // All input was fully consumed.
+        return AppLayerResult::ok();
+    }
+
+    fn on_request_gap(&mut self, _size: u32) {
```

Comment:
Do we need such a simple function with one unused arg ? instead of using the line directly